src/component/FavouriteTableRow.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {useContext, useState} from 'react';
+import {StyleSheet} from 'react-native';
+import styled from 'styled-components/native';
+import {DataTable} from 'react-native-paper';
+import Icon from 'react-native-vector-icons/FontAwesome5';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {UserContext} from '../context/User.Context';
+
+const EditText = styled.TextInput`
+  background-color: 'rgb(200,200,200)';
+  color: rgb(38, 38, 38);
+  flex: 1;
+  font-family: 'Montserrat-Regular';
+  font-size: 12px;
+  padding-bottom: 1px;
+  text-align: center;
+`;
+
+const ExerciseSetText = styled.Text`
+  color: rgb(38, 38, 38);
+  font-family: 'Montserrat-Regular';
+  font-size: 12px;
+  padding-bottom: 1px;
+  text-align: left;
+`;
+
+const WorkoutIcon = ({name}: {name: string}) => {
+  return <Icon name={name} size={14} color={'rgb(169,169,169)'} />;
+};
+
+export default function FavouriteTableRow({
+  row,
+  index,
+  workoutId,
+}: {
+  row: any;
+  index: Number;
+  workoutId: String;
+}) {
+  const {favouriteClickComplete, favouriteDeleteSet, favouriteEditSet} =
+    useContext(UserContext);
+
+  const [reps, setReps] = useState('10');
+  const [weight, setWeight] = useState('0');
+
+  const onChangedReps = (text: string) => {
+    isNaN(Number(text)) || text.includes('.')
```

Comment:
It's unclear what you are trying to do here

Reply:
Prevent user from being able to input partial reps or non-integer reps. Added comment above the const, as well as onChangedWeight which is similar